apps/api/src/services/boost-award.service.ts

```diff
@@ -22,19 +22,36 @@ type CompetitionPosition = {
   votingEndDate: Date;
 };
 
-const TEXT_ENCODER = new TextEncoder();
+type InitForStakeResult = BoostDiffResult & {
+  competitionId: string;
+  stakeId: bigint;
+};
+type InitNoStakeResult = BoostDiffResult & { competitionId: string };
 
-const DECIMALS = 18; // RECALL TOKEN DECIMALS
+const TEXT_ENCODER = new TextEncoder();
 
-class BoostAwardService {
+export class BoostAwardService {
+  readonly #db: Database;
+  readonly #competitionRepo: CompetitionRepository;
   readonly #boostRepository: BoostRepository;
+  readonly #stakesRepository: StakesRepository;
   readonly #userService: UserService;
-  readonly #divisor: bigint;
+  readonly #noStakeBoostAmount: bigint;
 
-  constructor(boostRepository: BoostRepository, userService: UserService) {
+  constructor(
+    database: Database,
+    competitionRepo: CompetitionRepository,
+    boostRepository: BoostRepository,
+    stakesRepository: StakesRepository,
+    userService: UserService,
+    noStakeBoostAmount?: bigint,
+  ) {
+    this.#db = database;
+    this.#competitionRepo = competitionRepo;
     this.#boostRepository = boostRepository;
+    this.#stakesRepository = stakesRepository;
     this.#userService = userService;
-    this.#divisor = 10n ** BigInt(DECIMALS);
+    this.#noStakeBoostAmount = noStakeBoostAmount || 1000n;
```

Comment:
why do we set a fallback amount here? if i'm reading the config correctly, if`NO_STAKE_BOOST_AMOUNT` env var is set, then `initNoStake` is used. it feels like the fallback value should happen in the config, and not the in this constructor. or, maybe not at all?

if you're in "stake" mode, this _technically_ doesn't matter. it's just a bit confusing.

i wonder if we could have a boolean env var `ENABLE_NO_STAKE_BOOST` dictate the "stake" vs "no-stake" behavior, or if that's overkill.

Reply:
Yea good call, I can make it optional and throw an error if it isn't set.